src/macro/eval/evalMisc.ml

```diff
@@ -284,7 +284,9 @@ let get_binop_fun op p = match op with
 	| OpShr -> op_shr p
 	| OpUShr -> op_ushr p
 	| OpMod -> op_mod p
-	| OpAssign | OpBoolAnd | OpBoolOr | OpAssignOp _ | OpInterval | OpArrow | OpIn -> die ~p "" __LOC__
+	| OpAssign | OpBoolAnd | OpBoolOr | OpAssignOp _ | OpInterval | OpArrow | OpIn ->
```

Comment:
This looks very weird. We aren't supposed to trigger any of these `die` situations, so having this kind of check here doesn't seem right.

We should instead add the relevant cases to the part which calls `get_binop_fun`. It already handles several of these, so the only ones left are `OpInterval`, `OpArrow` and `OpIn`.